### PR TITLE
[IOTDB-3043] AuthorInfo serialization and deserialization interfaces

### PR DESCRIPTION
**Pull request by RYH61**
AuthorInfo serialization and deserialization interfaces

### CODE REVIEW

File: node-commons/src/main/java/org/apache/iotdb/commons/utils/BasicStructureSerDeUtil.java [hunk 1]
```diff
@@ -239,4 +241,78 @@ public static Map<Integer, List<Integer>> readIntMapLists(ByteBuffer buffer) {
...
+    int length = readInt(buffer);
+    if (length == -1) {
+      return null;
+    }
+    List<String> strings = new ArrayList<>();
+    for (int i = 0; i < length; i++) {
+      strings.add(readString(buffer));
+    }
+    return strings;
+  }
```
Comment (CRZbulabula) [inline node-commons/src/main/java/org/apache/iotdb/commons/utils/BasicStructureSerDeUtil.java:317]: Please add tests in BasicStructureSerDeUtilTest.

File: server/src/main/java/org/apache/iotdb/db/auth/authorizer/BasicAuthorizer.java [hunk 2]
```diff
@@ -404,4 +410,105 @@ public void replaceAllUsers(Map<String, User> users) throws AuthException {
...
+          path = BasicStructureSerDeUtil.readString(buffer);
+        }
+        privilegeList.add(new PathPrivilege(privilegesId, path));
+      }
+      roleMap.put(roleName, new Role(roleName, privilegeList));
+    }
+
+    ((BasicUserManager) userManager).setUserMap(userMap);
+    ((BasicRoleManager) roleManager).setRoleMap(roleMap);
+  }
```
Comment (CRZbulabula) [inline server/src/main/java/org/apache/iotdb/db/auth/authorizer/BasicAuthorizer.java:513]: Please add serialize and deserialize tests for BasicAuthorizer.

**Review by CRZbulabula — CHANGES_REQUESTED** [review]
PTAL~

File: node-commons/src/main/java/org/apache/iotdb/commons/utils/BasicStructureSerDeUtil.java (hunk 1 above)
Comment (RYH61) [reply node-commons/src/main/java/org/apache/iotdb/commons/utils/BasicStructureSerDeUtil.java:317]: fix

File: server/src/main/java/org/apache/iotdb/db/auth/authorizer/BasicAuthorizer.java (hunk 2 above)
Comment (RYH61) [reply server/src/main/java/org/apache/iotdb/db/auth/authorizer/BasicAuthorizer.java]: fix

**Review by CRZbulabula — APPROVED** [review]
LGTM~

**Review by qiaojialin — COMMENTED** [review]
Could we directly use the file of .user and .role for the auth snapshot?

**Comment by RYH61** [comment]
> Could we directly use the file of .user and .role for the auth snapshot?
At present, when the system is started, the userMap has root information by default, but no other user information. When other users are used, they will be added to the user map. If you use these files directly, may not know which user information to restore.

**Pull request closed without merging**